### ISSUE TITLE
Jupyter Notebook not saved in History

### ISSUE THREAD

**Issue by SergeyYakubov** [galaxyproject/galaxy#14378]
**Describe the bug**

After I stop an Interactive Jupyter Notebook (e.g. the one shipped with the latest Galaxy - version="0.3" profile="22.01"), the dataset in History is green but empty.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: (22.05, version_minor:rc1)
Commit: (8b6030f67e62733afb6ae3bb2dd58cc7292e3104)

**To Reproduce**
1. Start an interactive notebook
2. Wait until it is running
3. Stop it
4. Click on view in History, or try to start the same tool but select "load previous notebook"

**Expected behavior**

an executed notebook should be available as an output dataset

**Additional context**

tried it on usegalaxy.eu as well - the same problem

**Comment by SergeyYakubov** [comment]
investigated some more - the behavior actually depends on the way one stops the notebook - if I got to File->Shut Down from the notebook - then it saves the notebook in the history as expected,

if I stop the notebook from Galaxy - User->Active InteractiveTools->Stop, then executed notebook dataset appears in history, is green, but empty. 

the problem seems to be in this [line](https://github.com/galaxyproject/galaxy/blob/e383e593e0d4e36c534a5ef9831bd7c44b023aa2/tools/interactive/interactivetool_jupyter_notebook.xml#L50): `jupyter lab --allow-root --no-browser --NotebookApp.shutdown_button=True && cp './${noteboook_name}.ipynb' '$jupyter_notebook'.` The `&& cp ...` part is never executed when a tool is stopped via Galaxy

So, this issue is not so critical and might even be not a bug, but a feature :), just this green empty Executed Notebook dataset is kind of confusing.

**Comment by mvdbeek** [comment]
> the problem seems to be in this [line](https://github.com/galaxyproject/galaxy/blob/e383e593e0d4e36c534a5ef9831bd7c44b023aa2/tools/interactive/interactivetool_jupyter_notebook.xml#L50): `jupyter lab --allow-root --no-browser --NotebookApp.shutdown_button=True && cp './${noteboook_name}.ipynb' '$jupyter_notebook'.` The `&& cp ...` part is never executed when a tool is stopped via Galaxy
> 
> So, this issue is not so critical and might even be not a bug, but a feature :), just this green empty Executed Notebook dataset is kind of confusing.

Yep, that's a good point, and I would probably call this a bug. I think we can solve this by using `from_work_dir="notebook.ipynb"`, which is guaranteed to run as the job is being finalized.

**Comment by mvdbeek** [comment]
> which is guaranteed to run as the job is being finalized.

I tried this, but we're actually just appending the `cp` command to the job template, so that also doesn't work.
